x-pack/solutions/security/plugins/security_solution/server/lib/detection_engine/prebuilt_rules/logic/rule_assets/prebuilt_rule_assets_client.ts

```diff
@@ -143,14 +143,25 @@ export const createPrebuiltRuleAssetsClient = (
           .map((v) => `(${attr}.rule_id: ${v.rule_id} AND ${attr}.version: ${v.version})`)
           .join(' OR ');
 
+        // Usage of savedObjectsClient.bulkGet() is ~25% more performant and
+        // simplifies deduplication but too many tests get broken.
+        // See https://github.com/elastic/kibana/issues/218198
         const findResult = await savedObjectsClient.find<PrebuiltRuleAsset>({
           type: PREBUILT_RULE_ASSETS_SO_TYPE,
           filter,
           perPage: MAX_PREBUILT_RULES_COUNT,
         });
 
         const ruleAssets = findResult.saved_objects.map((so) => so.attributes);
-        return validatePrebuiltRuleAssets(ruleAssets);
+        // Rule assets may have duplicates we have to get rid of.
+        // In particular prebuilt rule assets package v8.17.1 has duplicates.
+        const uniqueRuleAssets = new Map<string, PrebuiltRuleAsset>();
+
+        for (const ruleAsset of ruleAssets) {
+          uniqueRuleAssets.set(ruleAsset.rule_id, ruleAsset);
+        }
+
+        return validatePrebuiltRuleAssets(Array.from(uniqueRuleAssets.values()));
```

Comment:
Nit: this can be greatly simplified

```suggestion
        // Rule assets may have duplicates we have to get rid of.
        // In particular prebuilt rule assets package v8.17.1 has duplicates.
        const uniqueRuleAssets = uniqBy(ruleAssets, 'rule_id');

        return validatePrebuiltRuleAssets(uniqueRuleAssets);
```